internal/service/vmservice/utils.go

```diff
@@ -91,9 +136,13 @@ func shouldUpdateNetworkDevices(machineScope *scope.MachineScope) bool {
 		if len(net) == 0 {
 			return true
 		}
-		model, bridge := extractNetworkModelAndBridge(net)
+
+		model := extractNetworkModel(net)
+		bridge := extractNetworkBridge(net)
+		mtu := extractNetworkMTU(net)
+
 		// current is different from the desired spec.
-		if model != *v.Model || bridge != v.Bridge {
+		if model != *v.Model || bridge != v.Bridge || mtu != *v.MTU {
```

Comment:
Same?

Reply:
I have tested it.
This will set mtu to 1500 as it's the default value even when mtu is not defined in the spec.
probably separate it from the condition.